File: python_pkg/python_pkg/client_add_two_ints.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

def main(args = None):
    rclpy.init(args=args)
    node = Node("client_add_two_ints")
    client = node.create_client(AddTwoInts, "add_two_ints")

    while not client.wait_for_service(1):
        node.get_logger().warn("Waiting for Server add_two_ints")

    request = AddTwoInts.Request()
    request.a = 9
    request.b = 13
    for i in range(0, 500, 10):
        request.a += i
        request.b += i

        future = client.call_async(request)
        rclpy.spin_until_future_complete(node, future=future)

        try:
            response = future.result()
            node.get_logger().info(f"{request.a} + {request.b} = {response.sum}")

        except Exception as e:
            node.get_logger().error("Service call error %r" %(e,))


    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: cpp_packages/cpp_pkg/src/client_add_two_ints.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"



int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("client_add_two_ints");
    rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client;
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();


    client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");

    while(!client->wait_for_service(std::chrono::seconds(1))){
        RCLCPP_WARN(node->get_logger(), "Waiting for service");
    }

    request->a = 9;
    request->b = 80;

    for (int i = 0, j = 10; i <= 50; i += 1){
        request->a += j;
        request->b += j;

        auto future = client->async_send_request(request);
        if (rclcpp::spin_until_future_complete(node, future) == rclcpp::FutureReturnCode::SUCCESS)
            RCLCPP_INFO(node->get_logger(), "%d + %d = %d", 
                                            (int)request->a, (int)request->b, 
                                            (int)future.get()->sum);

        else
            RCLCPP_ERROR(node->get_logger(), "Something is not correct with the comunication");
        
    }


    rclcpp::shutdown();
    return 0;

}